tests/suites/test_suite_psa_crypto.function

```diff
@@ -3779,6 +3779,8 @@ void sign_fail( int key_type_arg, data_t *key_data,
     size_t signature_length = 0xdeadbeef;
     psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
 
+    TEST_ASSERT( signature_size != 0 );
```

Comment:
Why these assertions? If we wanted to test that `psa_asymmetric_sign` behaves properly when given an output buffer with a bad size, this test function should be the right thing to call. If something fails with `signature_size == 0` or `signature_size > PSA_ASYMMETRIC_SIGNATURE_MAX_SIZE`, the something in question should be fixed.

Reply:
This is due to coverity complaining on the unknown signature size. I added the same check that is made in other tests, to unify it.

Reply:
Ok, but in this particular test, `signature_size` is legitimate input. If Coverity is complaining about something, we need to look into why it's complaining, not silence it by pretending this can't happen.